### PR TITLE
chore(ci-visibility): add pytest Intelligent Test Runner support for unskippable tests and suites

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
Adds the ability to mark Pytest tests and suites (modules) as unskippable via the use of:

- `@pytest.mark.skipif(False, "datadog_itr_unskippable")` as a decorator around test functions
- `pytestmark = pytest.mark.skipif(False, reason="datadog_itr_unskippable")` at the module-level

This behavior is gated by the `DD_CIVISIBILITY_ITR_ENABLED` env var, without introducing a new feature flag.

Note: there is no release note because Intelligent Test Runner support for pytest is in beta

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/contrib/pytest/plugin.py [hunk 1]
```diff
@@ -431,9 +493,50 @@ def _get_test_class_hierarchy(item):
...
+
+        items_to_skip_by_module = {}
+        current_suite_has_unskippable_test = False
+
         for item in items:
+            suite_is_unskippable = _is_suite_unskippable(item)
+            test_is_unskippable = _is_test_unskippable(item)
+
+            if suite_is_unskippable:
+                log.debug("Module %s is marked as unskippable", item.module)
```
Comment (nikita-tkachenko-datadog) [inline ddtrace/contrib/pytest/plugin.py:505]: Should it, perhaps, be `log.debug("Suite %s in module %s is marked as unskippable", (item.name, item.module))`?

**Review by nikita-tkachenko-datadog — DISMISSED** [review]

File: ddtrace/contrib/pytest/plugin.py (hunk 1 above)
Comment (romainkomorndatadog) [reply ddtrace/contrib/pytest/plugin.py:505]: In `pytest`, we don't really have a difference between module and suite, so I think it'd be a little redundant, and in this case, `item.name` is a test.

Although, I guess probably more to the point, these debug logs don't appear at this stage of the process unless you bend over backwards and change the plugin's code to set up a logger before `pytest` does anyway, so for now, they're almost more like comments in the code than anything.

**Review by tonyredondo — DISMISSED** [review]

File: ddtrace/contrib/pytest/plugin.py (hunk 1 above)
Comment (romainkomorndatadog) [reply ddtrace/contrib/pytest/plugin.py]: Resolving since I removed `suite_is_unskippable`.

**Review by tonyredondo — APPROVED** [review]

**Review by nikita-tkachenko-datadog — APPROVED** [review]

File: ddtrace/contrib/pytest/plugin.py [hunk 2]
```diff
@@ -124,6 +134,37 @@ def _mark_not_skipped(item):
...
+
+def _mark_test_unskippable(test_item):
+    # type: (pytest.Test) -> None
+    test_span = _extract_span(test_item)
+    test_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+
+    suite_span = _extract_span(test_item.parent)
+    suite_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+
+    # module_item = _find_pytest_item(test_item, )
```
Comment (gnufede) [inline ddtrace/contrib/pytest/plugin.py:160]: nit: Commented code left behind?

File: ddtrace/contrib/pytest/plugin.py [hunk 3]
```diff
@@ -46,7 +47,7 @@
 from ddtrace.internal.logger import get_logger
 
 
-SKIPPED_BY_ITR = "Skipped by Datadog Intelligent Test Runner"
+SKIPPED_BY_ITR_REASON = "Skipped by Datadog Intelligent Test Runner"
```
Comment (gnufede) [inline ddtrace/contrib/pytest/plugin.py:50]: Since you're touching this, maybe move it to the constants file altogether?

**Review by gnufede — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Pull request merged**